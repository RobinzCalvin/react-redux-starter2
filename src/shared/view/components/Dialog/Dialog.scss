.dialog {
  $horizontal-padding: 0.8rem;

  display: inline-block;

  &__header {
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 0.5rem $horizontal-padding;
  }

  &__content {
    $border-style: 0.1rem solid #e5e5e5;

    padding: 1rem $horizontal-padding;
    border-top: $border-style;
    border-bottom: $border-style;
  }

  &__actions {
    padding: 0.5rem $horizontal-padding;
  }
}
