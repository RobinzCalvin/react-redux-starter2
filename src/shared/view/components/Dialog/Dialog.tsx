import * as React from 'react';
import block from 'bem-cn';
import Dialog from '@material-ui/core/Dialog';
import Typography from '@material-ui/core/Typography';

import './Dialog.scss';

interface IProps {
  title: string;
  isOpen: boolean;
  renderActions(): JSX.Element;
  onClose(): void;
}

const b = block('dialog');

class DialogComponent extends React.PureComponent<IProps> {
  public render() {
    const { title, isOpen, children, renderActions, onClose } = this.props;
    return (
      <Dialog open={isOpen} onClose={onClose}>
        <div className={b()}>
          <div className={b('header')}>
            <Typography variant="h6">{title}</Typography>
          </div>
          <div className={b('content')}>
            {children}
          </div>
          <div className={b('actions')}>
            {renderActions()}
          </div>
        </div>
      </Dialog>
    );
  }
}

export default DialogComponent;
