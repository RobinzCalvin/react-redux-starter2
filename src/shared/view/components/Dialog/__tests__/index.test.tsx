import React from 'react';
import { shallow } from 'enzyme';

import Dialog, { IDialogProps } from '../Dialog';

const props: IDialogProps = {
  title: 'title',
  onClose: jest.fn(),
  open: true,
};

describe('Dialog component', () => {
  const component = shallow(<Dialog {...props}><div/></Dialog>);
  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
