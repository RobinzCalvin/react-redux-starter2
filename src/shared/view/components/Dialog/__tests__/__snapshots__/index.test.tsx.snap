// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog component should match snapshot 1`] = `
<WithStyles(Dialog)
  onClose={[MockFunction]}
  open={true}
>
  <div
    className="dialog"
  >
    <div
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
    >
      <WithStyles(Typography)
        variant="h6"
      >
        title
      </WithStyles(Typography)>
    </div>
    <div />
  </div>
</WithStyles(Dialog)>
`;
